modules/afprog/afprog.c

```diff
@@ -449,7 +448,7 @@ afprogram_dd_store_reload_store_item(AFProgramDestDriver *self, GlobalConfig *cf
   reload_info->pid = self->pid;
   reload_info->writer = self->writer;
 
-  cfg_persist_config_add(cfg, afprogram_dd_format_persist_name(self), reload_info, afprogram_reload_store_item_destroy_notify, FALSE);
+  cfg_persist_config_add(cfg, afprogram_dd_format_persist_name(self), reload_info, (GDestroyNotify)afprogram_reload_store_item_destroy_notify, FALSE);
 }
```

Comment:
Try not to cast here please, again the function itself should match the prototype.

See the recent fix where an ack callback was casted and then its prototype changed, causing misterious errors.